main.js

```diff
@@ -31,21 +31,79 @@
 // default server
 // timezone
 
-// these are the 'global' values we always want to be available
-// you can also save/restore other values (and should, for device details)
+var contexts = [ 'page' ];
 
-var defaultStorage = {
-  tidepool: {
-    username: '',
-    password: '',
-    remember_me: false
+
+var contextMenus = [
+  {
+    type: 'normal',
+    id: 'MENUROOT',
+    title: 'Server',
+    contexts: contexts
+  },
+  {
+    type: 'radio',
+    id: 'Local',
+    title: 'Local',
+    contexts: contexts,
+    parentId: 'MENUROOT',
+    checked: false
+  },
+  {
+    type: 'radio',
+    id: 'Development',
+    title: 'Development',
+    contexts: contexts,
+    parentId: 'MENUROOT',
+    checked: false
+  },
+  {
+    type: 'radio',
+    id: 'Staging',
+    title: 'Staging',
+    contexts: contexts,
+    parentId: 'MENUROOT',
+    checked: false
   },
-  defaultServer: 'local',
-  timezone: 'America/Los_Angeles',
-  dexcomPortPattern: '/dev/cu.usbmodem.+',
-  FTDIPortPattern: '/dev/cu.usbserial.+',
-  forceDeviceIDs: []
-};
+  {
+    type: 'radio',
+    id: 'Production',
+    title: 'Production',
+    contexts: contexts,
+    parentId: 'MENUROOT',
+    checked: true
+  }
+];
+
+function setServer(window, info) {
+  var serverdata = {
+    Local: {
+      API_URL: 'http://localhost:8009',
+      UPLOAD_URL: 'http://localhost:9122',
+      BLIP_URL: 'http://localhost:3000'
+    },
+    Development: {
+      API_URL: 'https://devel-api.tidepool.io',
+      UPLOAD_URL: 'https://devel-uploads.tidepool.io',
+      BLIP_URL: 'https://blip-devel.tidepool.io'
+    },
+    Staging: {
+      API_URL: 'https://staging-api.tidepool.io',
+      UPLOAD_URL: 'https://staging-uploads.tidepool.io',
+      BLIP_URL: 'https://blip-staging.tidepool.io'
+    },
+    Production: {
+      API_URL: 'https://api.tidepool.io',
+      UPLOAD_URL: 'https://uploads.tidepool.io',
+      BLIP_URL: 'https://blip-ucsf-pilot.tidepool.io'
+    }
+  };
+
+  console.log('will use', info.menuItemId, 'server');
```

Comment:
There are two consoles -- one for index, and one for "background" -- which is main.js. This console message gets printed there.